### ISSUE TITLE
.obsidian.vimrc not loading

### ISSUE THREAD

**Issue by ssex-dev** [esm7/obsidian-vimrc-support#90]
Hi,

sadly I am not been able to load .obsidian.vimrc. The error in the console is:

Error loading vimrc file .obsidian.vimrc from the vault root TypeError: cmEditor.on is not a function
    at VimrcPlugin.defineFixedLayout (eval at <anonymous> (app.js:1), <anonymous>:815:18)
    at VimrcPlugin.readVimInit (eval at <anonymous> (app.js:1), <anonymous>:502:22)
    at eval (eval at <anonymous> (app.js:1), <anonymous>:439:70

Has anyone a idea how to fix this?
My .obsidian.vimrc file is located in the vault root and only contains the content from the readme. Some help would be highly appreciated!

**Comment by rykugur** [comment]
I have this issue every time I (re-)install Obsidian on a new computer. I've found that you need to have the `.obsidian.vimrc` file already in place when you **install** the add-on. If you added the file after add-on installation, you have to uninstall/reinstall the add-on completely.

Every now and then it'll also just randomly stop reading from the vimrc file; I have to again uninstall/reinstall the add-on.

**Comment by esm7** [comment]
1. Can you guys copy here the console from the very beginning (there's a debug log or two that I need from there to understand the case)?
2. What are your editor options? Default editing mode (under Obsidian's settings -> Editor) - live preview or source mode? Is "use legacy editor" turned on?

**Comment by rykugur** [comment]
1. I'll be sure to grab the logs the next time it happens.
2. In my case I had just installed obsidian and chose to enable live preview. Legacy editor defaults to off. So, to answer your question: live preview, legacy editor turned off.

**Comment by heyitschun** [comment]
Running into the same issue. This is what I'm getting:

```
Obsidian Developer Console
VM112:8074 Readwise Official plugin: setting interval to  43200000 milliseconds
VM125:690 loading url-into-selection
VM126:43 Loading Hotkeys++ plugin
VM139:441 Vimrc plugin: using CodeMirror 6 mode
VM139:493 Error loading vimrc file  from the vault root EISDIR: illegal operation on a directory, read
```

Default editing mode = Live Preview
Legacy editor = off

Using Obsidian Sync if that's relevant

**Comment by esm7** [comment]
I'm not sure it's the same case as the others reported, but here we see that the Vimrc file name is undefined in the settings.
So first, go to the plugin settings and fill in ".obsidian.vimrc" under "Vimrc file name".
And to make sure other users won't reach the same situation, I'll make sure that a default is applied more robustly.

**Comment by esm7** [comment]
Fixed in 0.6.2.
